File: fix.py
import os

files = [
    r'AsistenciaDesktop_v2\Views\LoginWindow.xaml',
    r'AsistenciaDesktop_v2\Views\AlumnosView.xaml',
    r'AsistenciaDesktop_v2\Views\UsuariosView.xaml',
    r'AsistenciaDesktop_v2\Views\MainShell.xaml'
]

for f in files:
    if os.path.exists(f):
        with open(f, 'rb') as file:
            content = file.read()
        
        # Replace the ASCII question mark 3F or whatever it is, or we decode with replace
        # Actually, let's just decode using 'latin1' because that's what PowerShell Get-Content -Raw did (ANSI)
        # and then write back as UTF-8
        
        # If it has a BOM, we can just read it normally if it's correct.
        
        # Let's just fix the strings we know are broken
        # It could be we wrote 'Gesti?n' as literal b'Gesti?n'
        
        content = content.replace(b'INICIAR SESIN', 'INICIAR SESIÓN'.encode('utf-8'))
        content = content.replace(b'INICIAR SESI?N', 'INICIAR SESIÓN'.encode('utf-8'))
        content = content.replace(b'Contrasea', 'Contraseña'.encode('utf-8'))
        content = content.replace(b'Contrase?a', 'Contraseña'.encode('utf-8'))
        content = content.replace(b'Gestin de', 'Gestión de'.encode('utf-8'))
        content = content.replace(b'Gesti?n de', 'Gestión de'.encode('utf-8'))
        content = content.replace(b'Men? principal', 'Menú principal'.encode('utf-8'))
        
        with open(f, 'wb') as file:
            file.write(content)

File: fix.cs
using System;
using System.IO;
using System.Text;

class Program {
    static void Main() {
        string[] files = { 
            @"AsistenciaDesktop_v2\Views\LoginWindow.xaml", 
            @"AsistenciaDesktop_v2\Views\AlumnosView.xaml", 
            @"AsistenciaDesktop_v2\Views\UsuariosView.xaml", 
            @"AsistenciaDesktop_v2\Views\MainShell.xaml" 
        };
        foreach (var f in files) {
            if (File.Exists(f)) {
                string content = File.ReadAllText(f, Encoding.UTF8);
                content = content.Replace("INICIAR SESIN", "INICIAR SESIÓN");
                content = content.Replace("INICIAR SESI?N", "INICIAR SESIÓN");
                content = content.Replace("Contrasea", "Contraseña");
                content = content.Replace("Contrase?a", "Contraseña");
                content = content.Replace("Gestin de Alumnos", "Gestión de Alumnos");
                content = content.Replace("Gesti?n de Alumnos", "Gestión de Alumnos");
                content = content.Replace("Gestión de Alumns", "Gestión de Alumnos");
                content = content.Replace("Gestión de Alumn?s", "Gestión de Alumnos");
                File.WriteAllText(f, content, Encoding.UTF8);
            }
        }
    }
}
